### PR TITLE
Split Sampler from Graph Store to Support New PyG Sampling API

### PR DESCRIPTION
**Pull request by alexbarghi-nv**
Resolves https://github.com/rapidsai/graph_dl/issues/47

### CODE REVIEW

**Review by eriknw — APPROVED** [review]
LGTM insofar as I understand it and how it touches PropertyGraphs.

Do you intend to keep the print statements in the tests?

Are the FIXMEs for this PR or future PRs?

File: python/cugraph/cugraph/structure/property_graph.py [hunk 1]
```diff
@@ -1326,6 +1334,11 @@ def __get_all_vertices_series(self):
         if epd is not None:
             vert_sers.append(epd[self.src_col_name])
             vert_sers.append(epd[self.dst_col_name])
+        if len(vert_sers) > 1 and not all(
+            cudf.api.types.is_dtype_equal(vert_sers[0].index.dtype, s.index.dtype)
+            for s in vert_sers
+        ):
+            vert_sers = [s.reset_index(drop=True) for s in vert_sers]
```
Comment (eriknw) [inline python/cugraph/cugraph/structure/property_graph.py:1341]: I believe these lines aren't needed for SG, and I removed them from https://github.com/rapidsai/cugraph/pull/2804.  Can you remove it and verify things still work for you?

Comment (alexbarghi-nv) [reply python/cugraph/cugraph/structure/property_graph.py]: > 

I removed them and confirmed the tests still pass.

Comment (alexbarghi-nv) [reply python/cugraph/cugraph/structure/property_graph.py]: The FIXMEs are for future PRs.  They're mostly referring to planned features that are not urgent.

Comment (alexbarghi-nv) [reply python/cugraph/cugraph/structure/property_graph.py]: And I removed the print statements.

File: python/cugraph/cugraph/gnn/pyg_extensions/data/cugraph_store.py [hunk 2]
```diff
@@ -510,7 +479,47 @@ def __get_renumbered_vertex_data_from_sample(self, nodes_of_interest):
...
+            }),
+            {
+                'blue_vertex': [0, 5],
+                'red_vertex': [3, 11],
+                'green_vertex': [2, 8]
+            }
+        Output: {
+                'blue__etype1__green': [0, 1],
+                'red__etype2__red': [1],
+                'red__etype3__blue': [0]
```
[the remark this answers is not in the document]
Comment (alexbarghi-nv) [reply python/cugraph/cugraph/gnn/pyg_extensions/data/cugraph_store.py:511]: No suffix is required; vertex types could be named basically anything.  The same goes for edge types.

File: python/cugraph/cugraph/gnn/pyg_extensions/data/cugraph_store.py [hunk 3]
```diff
@@ -419,60 +416,32 @@ def _subgraph(self, edge_types):
...
+        The ith element of each of array refers to the same vertex.
+
+        Example Input: [5, 2, 10, 11, 8]
+        Output: {'red_vertex': [5, 8], 'blue_vertex': [2], 'green_vertex': [10, 11]},
+                {'red_vertex': [0, 1], 'blue_vertex': [0], 'green_vertex': [0, 1]},
+                {
+                  'red_vertex': [[5.0, 2.0], [3.0, 5.0]],
+                  'blue_vertex': [[6.2, 2.1]],
+                  'green_vertex': [[5.9, 2.0], [3.0, 1.0]]
+                }
```
[the remark this answers is not in the document]
Comment (alexbarghi-nv) [reply python/cugraph/cugraph/gnn/pyg_extensions/data/cugraph_store.py:439]: 😎

**Review by rlratzel — APPROVED** [review]

**Comment by BradReesWork** [comment]
@gpucibot merge

**Pull request merged**